curator-client/src/main/java/org/apache/curator/ConnectionState.java

```diff
@@ -213,16 +213,14 @@ private synchronized void checkTimeouts() throws Exception
      * Return the current session id
      */
     public long getSessionId() {
-        long sessionId = -1;
-        if (isConnected()) {
-            try {
-                ZooKeeper zk = zooKeeper.getZooKeeper();
-                if (zk != null) {
-                    sessionId = zk.getSessionId();
-                }
-            } catch (Exception e) {
-                // Ignore the exception
+        long sessionId = 0;
```

Comment:
We also want to track the sessionId when the connection timed-out happened, also it would be useful if we can know what's the disconnected session id when the create/setData/getData is called. 

Any concern of removing the isConnected() check?